### PR TITLE
Fix up ALL the bugs in root-signing v2!

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

#### Ticket Link

Fixes

* Fixes key revocation when revoking the root keys from the snapshot and timestamp roles (that are now online) https://github.com/theupdateframework/go-tuf/issues/167
* Correctly sets the thresholds for root and targets (3 out of 5 keyholders) and for snapshot and timestamp (1 online key)
* Increases the expires for snapshot/timestamp to 3 weeks to allow a buffer for the biweekly signing
* Fixes the bug with adding signatures. `AddOrUpdateSignature` from go-tuf doesn't handle our placeholder signatures 

Supersedes https://github.com/sigstore/root-signing/pull/35

#### Release Note

```release-note

```

### CODE REVIEW

**Comment by asraa** [comment]
@SantiagoTorres would you like to review this? changes summarized above

File: cmd/tuf/app/init.go [hunk 1]
```diff
@@ -108,18 +109,30 @@ func InitCmd(ctx context.Context, directory, previous string, targets targetsFla
...
+	}
+	// Add online snapshot and timestamp keys with a shorter expiration of three weeks.
 	for role, keyRef := range map[string]string{"snapshot": snapshotRef, "timestamp": timestampRef} {
+		// Add new online keys
 		signerKey, err := pkeys.GetKmsSigningKey(ctx, keyRef)
 		if err != nil {
 			return err
 		}
-		if err := repo.AddVerificationKeyWithExpiration(role, signerKey.Key, time.Now().AddDate(0, 0, 14).UTC()); err != nil {
+		if err := repo.AddVerificationKeyWithExpiration(role, signerKey.Key, time.Now().AddDate(0, 0, 21).UTC()); err != nil {
```
Comment (trishankatdatadog) [inline cmd/tuf/app/init.go:140]: Why is 21 hardcoded?

File: cmd/tuf/app/init.go [hunk 2]
```diff
@@ -108,18 +109,30 @@ func InitCmd(ctx context.Context, directory, previous string, targets targetsFla
...
 		}
-		if err := repo.SetThreshold("root", threshold); err != nil {
+		if err := repo.SetThreshold(role, threshold); err != nil {
 			return err
 		}
 	}
 
-	// Add online snapshot and timestamp keys with a shorter expiration.
+	// Revoke old root keys used for snapshot and timestamp.
+	for _, role := range []string{"snapshot", "timestamp"} {
```
Comment (trishankatdatadog) [inline cmd/tuf/app/init.go:118]: The function of this FOR loop is not obvious, especially with the `tufKey.IDs()[0]`. Would you mind documenting a bit? Thanks!

File: cmd/tuf/app/init.go (hunk 1 above)
Comment (asraa) [reply cmd/tuf/app/init.go:140]: 3 weeks, `AddDate` only gives params for month, year, and days, so 21 days

Comment (trishankatdatadog) [reply cmd/tuf/app/init.go:140]: I see, ok, so hardcoding makes sense. Just doc it pls? 🙂

**Comment by asraa** [comment]
added more comments!

**Review by trishankatdatadog — APPROVED** [review]

**Comment by trishankatdatadog** [comment]
Approved FWIW but unfortunately I have no approval rights here 🙂

**Review by dlorenc — APPROVED** [review]

**Pull request merged**